python/hail/table.py

```diff
@@ -380,23 +380,47 @@ def count(self):
     def _force_count(self):
         return self._jt.forceCount()
 
-    @typecheck_method(caller=str, key_struct=nullable(expr_struct()), value_struct=nullable(expr_struct()))
-    def _select(self, caller, key_struct=None, value_struct=None):
-        if key_struct is None:
+    @typecheck_method(caller=str, key_struct=oneof(nullable(expr_struct()), exactly("default")), value_struct=nullable(expr_struct()))
+    def _select(self, caller, key_struct="default", value_struct=None):
+        def is_copy(ast, name):
```

Comment:
you don't need to do this -- you can just say `pair[1] is pair[2]`

Reply:
`mt.locus` is `mt.row_key.locus`

Reply:
Oh, nice!

Reply:
Actually, this doesn't work. I need it to detect when a field is being copied from an old field, but with a different name. That way, when a key field is renamed, _select won't do the last keyBy(newName), avoiding an uneccesary scan.

I realized this when test_order_by broke (!), because in`hl.utils.range_table(10).order_by(hl.desc('idx')).idx.collect()`, after the order_by the table has a key but isn't sorted by it, and the implementation of collect renames the key, triggering a keyBy(uid) which sorts. order_by is definitely a pain with ordered tables.